activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb

```diff
@@ -0,0 +1,125 @@
+module ActiveRecord
+  module ConnectionAdapters
+    module MySQL
+      module DatabaseStatements
+        # Returns an ActiveRecord::Result instance.
+        def select_all(arel, name = nil, binds = [], preparable: nil)
+          result = if ExplainRegistry.collect? && prepared_statements
+            unprepared_statement { super }
+          else
+            super
+          end
+          @connection.next_result while @connection.more_results?
+          result
+        end
+
+        # Returns a record hash with the column names as keys and column values
+        # as values.
+        def select_one(arel, name = nil, binds = [])
+          arel, binds = binds_from_relation(arel, binds)
+          @connection.query_options.merge!(as: :hash)
```

Comment:
`stmt.execute` calls `result.each` in the internal.
https://github.com/brianmario/mysql2/blob/0.4.4/ext/mysql2/statement.c#L384
Need to set `query_options` before `stmt.execute`.